extension/updateswh.js

```diff
@@ -211,34 +210,34 @@ function setupGitLabInstance(url, pattern, type) {
 // order is important: first match will be used!
 
 var forgehandlers = [{
-        pattern: /^https?:\/\/github.com\/[^\/]*\/[^\/]+/,
-        reject:  /^https?:\/\/github.com\/(features|marketplace|orgs|topics|collections|settings|([^\/]*\/[^\/]*\/search\?))/,
+        pattern: /^https?:\/\/github\.com\/[^\/]+\/[^\/]+/,
+        reject:  /^https?:\/\/github\.com\/(features|marketplace|orgs|topics|collections|settings|([^\/]+\/[^\/]+\/search\?))/,
         type: 'GitHub',
         handler: setupGitHub
     },
     {
-        pattern: /^https?:\/\/bitbucket.org\/[^\/]*\/[^\/]+/,
-        reject:  /^https?:\/\/bitbucket.org\/(dashboard\/|product\/|account\/signin)/,
+        pattern: /^https?:\/\/bitbucket\.org\/[^\/]+\/[^\/]+/,
+        reject:  /^https?:\/\/bitbucket\.org\/(dashboard\/|product\/|account\/signin)/,
         type: 'Bitbucket',
         handler: setupBitbucket
     },
     {
-        pattern: /^https?:\/\/gitlab.com\/[^\/]*\/[^\/]+/,
-        reject:  /^https?:\/\/gitlab.com\/explore\//,
+        pattern: /^https?:\/\/gitlab\.com\/[^\/]+\/[^\/]+/,
+        reject:  /^https?:\/\/gitlab\.com\/explore\//,
         type: 'GitLab',
         handler: setupGitLab
     },
     // hardcoded list of gitlab instances		     
     {
-        pattern: /^https?:\/\/(0xacab.org|gite.lirmm.fr|framagit.org|gricad-gitlab.univ-grenoble-alpes.fr)\/[^\/]*\/[^\/]+/,
-        reject:  /^https?:\/\/(0xacab.org|gite.lirmm.fr|framagit.org|gricad-gitlab.univ-grenoble-alpes.fr)\/users\/sign_in/,
+        pattern: /^https?:\/\/(0xacab\.org|gite\.lirmm\.fr|framagit\.org|gricad-gitlab\.univ-grenoble-alpes\.fr)\/[^\/]+\/[^\/]+/,
+        reject:  /^https?:\/\/(0xacab\.org|gite\.lirmm\.fr|framagit\.org|gricad-gitlab\.univ-grenoble-alpes\.fr)\/users\/sign_in/,
         type: 'GitLab instance',
         handler: setupGitLabInstance
     },
     // heuristic: we handle gitlab.*.* as a GitLab instance
     {
-        pattern: /^https?:\/\/gitlab.[^.]*.[^.\/]*\/[^\/]*\/[^\/]+/,
-        reject:  /^https?:\/\/gitlab.[^.]*.[^.\/]*\/users\/sign_in/,
+        pattern: /^https?:\/\/gitlab\.[^.\/]+\.[^.\/]+\/[^\/]+\/[^\/]+/,
+        reject:  /^https?:\/\/gitlab\.[^.\/]+\.[^.\/]+\/users\/sign_in/,
```

Comment:
For example the regexp would incorrectly accept `'https://gitlab.com/remram44/remram.fr/tree/master'` (the first `[^.]*` would match `com/remram44/remram`)

Reply:
Thanks a lot @remram44  for spotting and fixing all this!